tests/test_tractseg.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import numpy as np
+import pytest
+import scipy.linalg
+
+from abcdmicro.dwi import Dwi
+from abcdmicro.resource import (
+    InMemoryBvalResource,
+    InMemoryBvecResource,
+    InMemoryResponseFunctionResource,
+    InMemoryVolumeResource,
+)
+from abcdmicro.tractseg import extract_tractseg
+
+
+@pytest.fixture
+def affine_random():
+    rng = np.random.default_rng(7562)
+    affine = np.eye(4, dtype=float)
+    affine[:3, 3] = 100 * (
+        rng.random(size=(3,), dtype=float) - 0.5
+    )  # random translation
+    random_3by3 = rng.random(size=(3, 3), dtype=float) - 0.5
+    affine[:3, :3] = scipy.linalg.expm(
+        random_3by3 - random_3by3.T
+    )  # exponentiate a random skew-symmetric matrix to get some orthogonal matrix
+    return affine
+
+
+@pytest.fixture
+def dwi_data_small_random(affine_random) -> Dwi:
+    rng = np.random.default_rng(1337)
+    dwi_data = rng.random(size=(3, 4, 5, 6))
+    bvals = np.array([0, 1000, 500, 0, 0, 500], dtype=float)
+    bvecs = rng.random(size=(6, 3))
+    bvecs = bvecs / np.sqrt((bvecs**2).sum(axis=1, keepdims=True))
+    return Dwi(
+        volume=InMemoryVolumeResource(array=dwi_data, affine=affine_random),
+        bval=InMemoryBvalResource(bvals),
+        bvec=InMemoryBvecResource(bvecs),
+    )
+
+
+@pytest.fixture
+def response_function() -> InMemoryResponseFunctionResource:
+    rng = np.random.default_rng(13)
+    sh_coeffs = rng.random(size=(10,))  # random array of coeeficients
+    avg_signal = rng.random(size=(), dtype=np.float32)
+    return InMemoryResponseFunctionResource(sh_coeffs=sh_coeffs, avg_signal=avg_signal)
+
+
+@pytest.mark.parametrize("response", [None, response_function])
+def test_tractseg(mocker, dwi_data_small_random, response):
```

Comment:
I think the `response_function` you have here is the _python function_ known as `response_function` (a pytest fixture defined above), rather than the return value of it (which is a response function in the sense of being a `InMemoryResponseFunctionResource`). Here is a possible easy fix:

```suggestion
@pytest.mark.parametrize("provide_response", [False, True])
def test_tractseg(mocker, dwi_data_small_random, response_function, provide_response):
    response = response_function if provide_response else None
```